lib/dry/web/roda/generators/abstract_project.rb

```diff
@@ -0,0 +1,52 @@
+require "dry/web/roda/generate"
+
+module Dry
+  module Web
+    module Roda
+      module Generators
+        class AbstractProject
+          attr_reader :target_dir, :options
+
+          def initialize(target_dir, options = {})
+            @target_dir = target_dir
+            @options = options
+            @generator = Generate.new(destination)
+            @templates = []
```

Comment:
We could even freeze the templates array after this is done.

Of course, we could restructure these internal methods to avoid mutating an ivar but I think the readability here is fine and if we know they're intended to be run only when the object is initialized I think it's fine :)